public/app/plugins/datasource/cloud-monitoring/components/Experimental/Metrics.tsx

```diff
@@ -0,0 +1,184 @@
+import { css } from '@emotion/css';
+import { startCase, uniqBy } from 'lodash';
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { GrafanaTheme2, SelectableValue } from '@grafana/data';
+import { EditorRow, EditorField, EditorFieldGroup } from '@grafana/experimental';
+import { getSelectStyles, Select, useStyles2, useTheme2 } from '@grafana/ui';
+
+import CloudMonitoringDatasource from '../../datasource';
+import { MetricDescriptor } from '../../types';
+
+export interface Props {
+  refId: string;
+  onChange: (metricDescriptor: MetricDescriptor) => void;
+  templateVariableOptions: Array<SelectableValue<string>>;
+  datasource: CloudMonitoringDatasource;
+  projectName: string;
+  metricType: string;
+  children: (metricDescriptor?: MetricDescriptor) => JSX.Element;
+}
+
+interface State {
+  metricDescriptors: MetricDescriptor[];
+  metrics: any[];
+  services: any[];
```

Comment:
Think both `metrics` and `services` should be of type `Array<SelectableValue<string>>`